### PR TITLE
feat(lsp): update lsp import + fix incompatibilities

### PR DESCRIPTION
**Pull request by grouville**
- Change hidden Dagger subcommand to `dagger cuelsp`.
- Update LSP imports
- Update server Mode initialization to work with functional server initialization paradigm + server initialization error management

Signed-off-by: guillaume

### CODE REVIEW

File: cmd/dagger/cmd/lsp.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 var lspCmd = &cobra.Command{
-	Use:    "lsp",
+	Use:    "cuelsp",
 	Short:  "Run Dagger CUE Language Server",
 	Hidden: true,
```
Comment (dolanor) [inline cmd/dagger/cmd/lsp.go:16]: Why do we hide it?

File: cmd/dagger/cmd/lsp.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 var lspCmd = &cobra.Command{
-	Use:    "lsp",
+	Use:    "cuelsp",
 	Short:  "Run Dagger CUE Language Server",
```
Comment (dolanor) [inline cmd/dagger/cmd/lsp.go:15]: Should we document that we need to press 3 times CTRL+C to stop it?
It's not classical to do that, so I guess we should document it.
Or at least log it when the first one is pressed that we're shutting down nicely or that we need 2 more to stop the server.

We could put it in the long help description of it.

**Review by dolanor — APPROVED** [review]
LGTM
Tested locally, and it works.
The other comments are not blocking, just things to remember. Create issues about it to track if not handled here.

File: cmd/dagger/cmd/lsp.go (hunk 1 above)
Comment (grouville) [reply cmd/dagger/cmd/lsp.go:16]: It's gonna be easier to manage the future legacy of that subcommand

File: cmd/dagger/cmd/lsp.go (hunk 2 above)
Comment (grouville) [reply cmd/dagger/cmd/lsp.go:15]: Noted, thanks. Not a blocker, you're right

**Pull request merged**